packages/cli/src/cli/cogify/action.cog.ts

```diff
@@ -45,25 +46,34 @@ export class ActionCogCreate extends CommandLineAction {
           'Failed to find cog name from batch index',
         );
       }
-      return name;
+      return [name];
     }
 
-    const cogIndex = this.cogIndex?.value;
-    if (cogIndex != null) {
-      const { name } = files[cogIndex];
-      if (name == null) {
-        throw new LoggerFatalError({ cogIndex, tileMax: files.length - 1 }, 'Failed to find cog name from index');
+    const cogIndex = this.cogIndex?.values;
+    if (cogIndex != null && cogIndex.length > 0) {
+      const names: string[] = [];
+      for (const i of cogIndex) {
+        const { name } = files[i];
```

Comment:
This is kinda weirrd, if i > files.length, files[i]  will be undefined and  will throw

```typescript
const file = files[i]
if (files == null) throw new LoggerFatalError(...)
names.push(file.name);
```

Reply:
might also be a good idea to use a `Set` then return `[...set.values()]` as it will dedupe files.